### ISSUE TITLE
some questions about _neg_loss

### ISSUE THREAD

**Issue by feifeiwei** [princeton-vl/CornerNet#69]
def _neg_loss(preds, gt):
    pos_inds = gt.eq(1)
    neg_inds = gt.lt(1)

    neg_weights = torch.pow(1 - gt[neg_inds], 4)

    loss = 0
    for pred in preds:
        pos_pred = pred[pos_inds]
        neg_pred = pred[neg_inds]

        pos_loss = torch.log(pos_pred) * torch.pow(1 - pos_pred, 2)
        neg_loss = torch.log(1 - neg_pred) * torch.pow(neg_pred, 2) * neg_weights

        num_pos  = pos_inds.float().sum()
        pos_loss = pos_loss.sum()
        neg_loss = neg_loss.sum()

        if pos_pred.nelement() == 0:
            loss = loss - neg_loss
        else:
            loss = loss - (pos_loss + neg_loss) / num_pos
    return loss

 loss = loss - (pos_loss + neg_loss) / num_pos

I found the neg_loss is also divided by num_pos.
why does the neg_loss not divided by num_neg????

**Comment by heilaw** [comment]
We follow the focal loss paper, which also divides the loss by the number of positive anchors instead of the total number of anchors.
